flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SsgNetworkMemoryCalculationUtils.java

```diff
@@ -93,28 +93,27 @@ public static void enrichNetworkMemory(
     private static TaskInputsOutputsDescriptor buildTaskInputsOutputsDescriptor(
             ExecutionJobVertex ejv, Function<JobVertexID, ExecutionJobVertex> ejvs) {
 
-        Map<IntermediateDataSetID, Integer> maxInputChannelNums;
-        Map<IntermediateDataSetID, Integer> maxSubpartitionNums;
+        Map<IntermediateDataSetID, Integer> maxInputChannelNums = new HashMap<>();
```

Comment:
It's a bit weird that the getXXX function returns void but fulfills these info to the argument passed in. How about adding something like:
```
TaskInputsInfo
	int maxInputChannelNums
TaskOutputsInfo
	int maxSubpartitionNums;
  ResultPartitionType partitionType;
```

Then let getMaxInputChannelNums/getMaxSubpartitionNums return `Map<IntermediateDataSetID, TaskInputsInfo/TaskOutputsInfo>`. And the name may need to be changed to getTaskInputsInfo/getOutputsInfo.

In this way, we could add inputPartitionTypes to TaskInputsInfo in the future.

Reply:
Thanks for helping reivew and Sorry about the late reply. @huwh 
But the PR has been closed at https://issues.apache.org/jira/browse/FLINK-30473.
I agree that your idea makes sense, but I think it is also not a serious issue and may refactor it later.